### ISSUE TITLE
Make OpenXR library a fetch dependency instead

### ISSUE THREAD

**Issue by rbessems** [KhronosGroup/OpenXR-Tutorials#1]
Instead of describing how to download the OpenXR SDK manually and set various paths in CMake should we use this automatic fetched dependency?

```OPTION(USE_OPENXR_LAYERS "Use OpenXR layers" ON)
set(BUILD_ALL_EXTENSIONS ON CACHE INTERNAL "Build loader and layers with all extensions")
set(BUILD_TESTS OFF CACHE INTERNAL "Build tests")
set(BUILD_CONFORMANCE_TESTS OFF CACHE INTERNAL "Build conformance tests")
set(BUILD_API_LAYERS ${USE_OPENXR_LAYERS} CACHE INTERNAL "Use OpenXR layers")
FetchContent_Declare(
        oxr
        URL_HASH MD5=390455e9395a92d6b730afb8f23fac8e
        URL openxr_source_sdk,https://github.com/KhronosGroup/OpenXR-SDK-Source/archive/refs/tags/release-1.0.27.zip)```
```

Then the application can use this simply via:

```
target_link_libraries(openxrtutorialch2 PUBLIC
        openxr_loader
```

EDIT: Doing this will remove a lot of explanation code from the tutorial on how to build and install the OpenXR SDK which is just distracting from the purpose.

**Comment by AndrewRichards-Code** [comment]
Commit 05199b668aa44e6fca68cc4f917ac4fa4580951c unifies the CMake for PC and Android.
app/build.gradle is set up like this at the moment, and we use `FetchContent_Declare()` in the CMake.
```
android {
    //[...]
    // enable prefab support for the OpenXR AAR
    //buildFeatures {
    //    prefab true
    //}
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'androidx.appcompat:appcompat:1.0.2'
    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
    implementation 'org.khronos.openxr:openxr_loader_for_android:1.0.27'
    //implementation files('../../thirdparty/openxr_loader_manifest_only-1.0.27.aar')
}
```
I have tried @rpavlik `openxr_loader_manifest_only-1.0.27.aar`, but I am getting this build error:
```
> Task :app:processDebugResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform openxr_loader_manifest_only-1.0.27.aar to match attributes {artifactType=android-compiled-dependencies-resources}.
      > Execution failed for AarResourcesCompilerTransform: C:\Users\Andrew\.gradle\caches\transforms-3\9454c3d3ed5ad0b48d4d7553f7fc8985\transformed\jetified-openxr_loader_manifest_only-1.0.27.
         > C:\Users\Andrew\.gradle\caches\transforms-3\9454c3d3ed5ad0b48d4d7553f7fc8985\transformed\jetified-openxr_loader_manifest_only-1.0.27\AndroidManifest.xml
```
I'm not sure what could be going on here.

**Comment by rpavlik** [comment]
that's weird, can you share the files it's mentioning? (Also, do you really need jetifier enabled?)

**Comment by AndrewRichards-Code** [comment]
Commit 43959909c68c148400eb5a27a30a81dfc80856af disables jetifier.
So the updated output is:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform openxr_loader_manifest_only-1.0.27.aar to match attributes {artifactType=android-compiled-dependencies-resources}.
      > Execution failed for AarResourcesCompilerTransform: C:\Users\Andrew\.gradle\caches\transforms-3\539f688e41ddd31f6e22278ca69808cd\transformed\openxr_loader_manifest_only-1.0.27.
         > C:\Users\Andrew\.gradle\caches\transforms-3\539f688e41ddd31f6e22278ca69808cd\transformed\openxr_loader_manifest_only-1.0.27\AndroidManifest.xml
```
The `.aar` that I am using just contains the `AndroidManifest.xml`:
https://github.com/KhronosGroup/OpenXR-Tutorials/blob/main/thirdparty/openxr_loader_manifest_only-1.0.27.aar
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="org.khronos.openxr.loader_for_android"
  android:versionCode="1"
  android:versionName="1.0.27">
  <!--
  Copyright (c) 2020 The Khronos Group Inc.
  SPDX-License-Identifier: Apache-2.0
  -->

  <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="29"/>

  <uses-permission android:name="org.khronos.openxr.permission.OPENXR" />

  <queries>
    <provider android:authorities="org.khronos.openxr.runtime_broker;org.khronos.openxr.system_runtime_broker" />
  </queries>

</manifest>
```
It appears that Gradle is looking in the wrong directory. There should be an extra `openxr_loader_manifest_only\` on the path.

C:\Users\Andrew\.gradle\caches\transforms-3\539f688e41ddd31f6e22278ca69808cd\transformed\openxr_loader_manifest_only-1.0.27\
|-- `jars\`
|-- `openxr_loader_manifest_only-1.0.27\`
|-- |-- `META-INF\`
|-- |-- `AndroidManifest.xml`

**Comment by AndrewRichards-Code** [comment]
I am not too familiar with the finer details of the Android/Gradle build system. Is this too much for a tutorial?

**Comment by rpavlik** [comment]
I might see if I can take a peek at it. The aar file is just a zip file renamed so it's easy to tinker with. Trying to make it so that you just add the line to the gradle file and be done :)

**Comment by rpavlik** [comment]
> FYI, I have an MR pending in gitlab to allow you to use `OpenXR::openxr_loader` as the library name whether you're vendoring the source or building against found binaries. https://gitlab.khronos.org/openxr/openxr/-/merge_requests/2793

OK this is released in 1.0.28.
